helpers/jobScheduler.js

```diff
@@ -0,0 +1,29 @@
+
+var date = require('date-fns')
+const db = require('../config/database');
+var Agenda = require('agenda')
+
+async function setTimer (timestamp, id, callbackArgs, callback) {
+	if(!date.isValid(new Date(timestamp))){
+		console.error("Invalid timestamp in scheduler:" + timestamp)
+		return;
+	}
+
+	const agenda = new Agenda({db: {address: db.mongoURI}});
+
+	//wait for the define to process
+	await new Promise(resolve => agenda.once('ready', resolve));
+
+	//cancel any current timeouts with this id
+	await agenda.cancel({name: id}, (err, numRemoved) =>{
+		if(err)console.error(err)
```

Comment:
What bad things can happen if we have multiple timeouts on the same user? Would we maybe want to bubble up an error to the app and make the user try again on some of them?

Reply:
We have to cancel all previous timeouts for that don so that the don is able to edit their clockOut time. For example, a don goes on and sets a clock out time of 8pm Saturday. Another don bails, so the decide to change it to 8:30am Monday. If the original timeout is not cancelled, the don will be turned off on Saturday even if they editted the time.

Admittedly there might be a better way to do this since there is a way to make jobs unique, but that feature seemed poorly documented and this works just fine.

Bubbling up the error is a good idea, I can look into adding that. Maybe another field in donInfo which includes error information if the timeout failed, since failed timeout would be a really annoying bug to detect.